internal/backends/sqlite/database.go

```diff
@@ -95,6 +95,26 @@ func (db *database) DropCollection(ctx context.Context, params *backends.DropCol
 	return nil
 }
 
+// Ping implements backends.Database interface.
+func (db *database) Ping(ctx context.Context) error {
+	sqlDB := db.r.DatabaseGetExisting(ctx, db.name)
+
+	// create temporary non-existent database to make sure that it's possible for user to create one
+	// (there might be some fs permission errors)
+	if sqlDB == nil {
+		var err error
+
+		sqlDB, err = db.r.DatabaseGetOrCreate(ctx, db.name)
+		if err != nil {
+			return lazyerrors.Error(err)
+		}
+
+		defer db.r.DatabaseDrop(ctx, db.name)
```

Comment:
Here I have my favorite question - what if someone worked with a database with this name in another request (e.g. inserted some data)? In this case, we will drop it. 

I understand why you do it - there are not many commands that would cause interaction with the database, but this particular approach doesn't really look safe. What do you think?

Reply:
Is there transaction equivalent which ensures rollback? Rollback on transaction is more solid than defer drop. :hugs:

Reply:
Those are good points – ping should not cause any observable side-effects, and creating even a temporary database or collection is observable by a parallel client